app/utilities/UserHelper.java

```diff
@@ -0,0 +1,148 @@
+package utilities;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import models.Address;
+import models.Token;
+import models.User;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.http.ParseException;
+import play.libs.Json;
+
+
+/**
+ * Created by test on 03/02/16.
+ */
+public class UserHelper {
+
+    private User user;
+
+    public UserHelper(JsonNode json){
+        System.out.println(">>>>>>json " + json);
+        user = Json.fromJson(json, User.class);
+    }
+
+    /**
+     * Creates a User with specified information. Password is hashed using MD5.
+     * Finally, confirmation mail is sent to specified email address.
+     */
+    public void createUser(){
+        user.setPassword(DigestUtils.md5Hex(user.getPassword()));
+        user.setPasswordConfirmation(DigestUtils.md5Hex(user.getPasswordConfirmation()));
+        Email.sendConfirmationEmail(user.getEmail(), Resources.SERVER_NAME + "/" + Resources.VERSION
+                + "/registration/confirm/" + encodeToken(user.getAuthToken().getToken()));
+        user.save();
+    }
```

Comment:
I would prefer creating general DataPersistenceManager for handling all database specific calls. This helper class can be used for some User specific operations (like generating token).